### ISSUE TITLE
Add support for Exif 2.32

### ISSUE THREAD

**Issue by boardhead** [Exiv2/exiv2#929]
I just wanted to let you know that version 2.32 of the Exif specification was released last month.  This version adds a few new tags that should be supported by Exiv2:

https://web.archive.org/web/20190624045241if_/http://www.cipa.jp:80/std/documents/e/DC-008-Translation-2019-E.pdf

I checked the Exiv2 code base to see what version is currently supported, but I couldn't find any of the new tags from Exif 2.31 (July 2016) or Exif 2.3 (April 2010) either.  I hope I am missing something.  There were a number of useful tags added in these versions as well.

**Comment by clanmills** [comment]
Thanks for letting us know.  I thought Andreas Huggel had dealt with 2.31. 

I'll investigate and add the tags.  I believe that's straightforward and nothing likes as complicated as adding support for the FocusPosition tag in Sony Raw #582.

**Comment by clanmills** [comment]
I've had a look at the spec.
```

2.3 (Revised 2012) | December 2012 | Published Revised Revision 2.3 CorrectedanexplanationpartofOrientationTag Corrected an explanation part of GPS Status Tag
2.31 | July 2016 | Published Revision 2.31 Added time difference to UTC(Universal Time Coordinated) astags relating to Date and Time- Added three time offset tags respectively correspondingto the three existing tags (File change date and time, Date and time of original data generation, Date and time of digital data generation) Added tags relating to shooting situation(Temperature, Humidity, Pressure, WaterDepth, Acceleration, CameraElevationAngle) Corrected misprints and omissions throughout the text (Including the reflection of the contents of the corrigendum established in September 2014 to the main text)
2.32 | May 2019 | Published Revision 2.32 Added 3 tags relating to a composite image- CompositeImage, SourceImageNumberOfCompositeImage, SourceExposureTimesOfCompositeImage Renumbered according to the addition of figures.
```
Andreas update Exiv2 for 2.3 about 2013 or so.  I didn't know about 2.31 and now 2.32.  I'll try to deal with this for v0.27.3 (scheduled 2019-09-30).

**Comment by boardhead** [comment]
You referenced only the 2.3 revision.  The original 2.3 spec was April 2010, and it added a bunch of new tags.

**Comment by clanmills** [comment]
I was confused.  The document says: _Established on April, 2010_ and the update history says _2.3 (Revised 2012)_.  Andreas updated Exiv2 for 2.3.

Was it for 2.3 2010, or 2.3 2012?  I can't remember.  I'll investigate and bring Exiv2 up-to-date.

**Comment by clanmills** [comment]
Pushing this to Exiv2 v0.27.4 (scheduled 2019-12-31).

**Comment by clanmills** [comment]
Gosh, those standard documents are difficult to read.  I think Exiv2 is implements 2.3.  I've found a tag that wasn't implemented:

```
+        TagInfo(0x001f, "GPSHPositioningError", N_("GPS Horizontal positioning error"),
+                N_("This tag indicates horizontal positioning errors in meters."),
+                gpsId, gpsTags, unsignedRational, 1, print0x001e),
```

The 2.32 spec has a better revision statement:

-------------

Items that underwent additions of tag specifications for this revision are as follows:

 1. [Exif IFD] CompositeImage (Recommended)
- Indicates whether the recorded image is a composite image or not.
2. [Exif IFD] SourceImageNumberOfCompositeImage (Optional)
- Indicates the number of the source images (tentatively recorded images) captured for a composite Image.
3. [Exif IFD] SourceExposureTimesOfCompositeImage (Optional)
- For a composite image, records the parameters relating exposure time of the exposures for
generating the said composite image, such as respective exposure times of captured source images (tentatively recorded images).

2.31 | July 2016 | Published Revision 2.31

Added time difference to UTC(Universal Time Coordinated) as tags relating to Date and Time

Added three time offset tags respectively correspondingto the three existing tags (File change date and time, Date and time of original data generation, Date and time of digital data generation)

Added tags relating to shooting situation (Temperature, Humidity, Pressure, WaterDepth, Acceleration, CameraElevationAngle)

Corrected misprints and omissions throughout the text (Including the reflection of the contents of the corrigendum established in September 2014 to the main text)

-------------

**Comment by clanmills** [comment]
Done.  Thanks to @kmilos for very good contribution and comments on this topic.  And the effort is reflected all the way into the book.  https://clanmills.com/exiv2/book/#Exif
